go.mod

```diff
@@ -1,74 +1,81 @@
 module github.com/tailscale/caddy-tailscale
 
-go 1.21
+go 1.22.0
+
+toolchain go1.22.1
```

Comment:
we generally leave off the toolchain requirements, since we seldom have dependencies on specific toolchain version.  I'll go ahead and update that.

Reply:
`go get -u` made this change, so I chose to trust the tool. Noted for the future :D